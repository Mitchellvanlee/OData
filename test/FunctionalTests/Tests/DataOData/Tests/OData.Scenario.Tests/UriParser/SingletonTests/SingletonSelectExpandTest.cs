//---------------------------------------------------------------------
// <copyright file="SingletonSelectExpandTest.cs" company="Microsoft">
//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
// </copyright>
//---------------------------------------------------------------------

namespace Microsoft.Test.Taupo.OData.Scenario.Tests.UriParser
{
    using System.Runtime.CompilerServices;
    using Microsoft.VisualStudio.TestTools.UnitTesting;

    [TestClass]
    public class SingletonSelectExpandTest : UriParserTestsBase
    {
        [TestMethod]
		[MethodImplAttribute(MethodImplOptions.NoOptimization)]
        public void SingleNavPropSelectExpandTest()
        {
            this.ApprovalVerifySelectAndExpandParser(specialOrderBase, "", "CustomerForOrder");
        }

        [TestMethod]
		[MethodImplAttribute(MethodImplOptions.NoOptimization)]
        public void TwoNapPropsExpanded()
        {
            this.ApprovalVerifySelectAndExpandParser(specialOrderBase, "", "CustomerForOrder,OrderDetails");
        }

        [TestMethod]
		[MethodImplAttribute(MethodImplOptions.NoOptimization)]
        public void ExpandWhenPathSelected()
        {
            this.ApprovalVerifySelectAndExpandParser(specialOrderBase, "", "CustomerForOrder($select=*)");
        }

        public void ExpandWhenItemPrimitiveAndNavPropSelected()
        {
            this.ApprovalVerifySelectAndExpandParser(specialOrderBase, "OrderDate,OrderDetails", "OrderDetails");
        }

        [TestMethod]
		[MethodImplAttribute(MethodImplOptions.NoOptimization)]
        public void SelectPrimitiveOnExpandedNavProp()
        {
            this.ApprovalVerifySelectAndExpandParser(specialOrderBase, "", "OrderDetails($expand=ProductOrdered($select=QuantityInStock))");
        }

        

    }
}
