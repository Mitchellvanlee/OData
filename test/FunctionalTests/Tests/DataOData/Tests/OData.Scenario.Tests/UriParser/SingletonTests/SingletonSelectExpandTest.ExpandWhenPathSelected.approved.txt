$select = 
$expand = CustomerForOrder($select=*)


SelectExpandQueryOption
	AllSelected = True
	SelectedItems
		Expanded Navigation Property
			Path[(NavigationProperty: CustomerForOrder)]
			SelectExpandQueryOption
				AllSelected = False
				SelectedItems
					(Wildcard)